### PR TITLE
If you like to...

### LINKED ISSUE #58: The UserEndpoint doesn't work well.
```
Deezer deezer = DeezerSession.CreateNew();
if (!deezer.IsAuthenticated) {
      await deezer.Login(access_token);
}
IUser user = deezer.Browse.CurrentUser.User;
```
The user object is empty. The iClient Object is null.

IEnumerable<ITrack> tracks = await user.First().GetFavouriteTracks();
gets an exception:
```
NullReferenceException: Object reference not set to an instance of an object.
E.Deezer.Api.User.Get<TSource, TDest>(string aMethod, DeezerPermissions aPermisisons, uint aStart, uint aCount) in User.cs
+
            if (!Client.IsAuthenticated)
```

### PR DESCRIPTION
**Pull request by tolbxela**
Fixes #58

### CODE REVIEW

File: E.Deezer/Api/User.cs [hunk 1]
```diff
@@ -8,13 +8,29 @@ namespace E.Deezer.Api
...
 	public interface IUser : IObjectWithImage
 	{
-		ulong Id { get;  }
-		string Name { get;  }
-		string Link { get;  }
-		string Country { get;  }
-
-
-		// ** Methods **
+        ulong id { get; set; }
```
Comment (projectgoav) [inline E.Deezer/Api/User.cs:11]: Could we have properties starting with a captial letter please? It'll match the rest of the API that way.

File: E.Deezer/DeezerClient.cs [hunk 2]
```diff
@@ -114,31 +115,31 @@ public Task<IChart> GetChart(ulong aId, uint aStart, uint aCount)
...
         {
-            return iExecutor.ExecuteGet(aMethod, RequestParameter.EmptyList)
+            if (aParams == null) aParams = RequestParameter.EmptyList;
+            return iExecutor.ExecuteGet(aMethod, aParams)
                             .ContinueWith(t =>
                             {
                                 CheckHttpResponse(t);
                                 T deserialized = DeserializeResponse<T>(t.Result.Content).Result;
-
+                                var msg = t.Result.Content.ReadAsStringAsync().Result;
```
Comment (projectgoav) [inline E.Deezer/DeezerClient.cs:127]: `msg` doesn't appear to be used. Think that could go!

File: E.Deezer/DeezerClient.cs [hunk 3]
```diff
@@ -114,31 +115,31 @@ public Task<IChart> GetChart(ulong aId, uint aStart, uint aCount)
...
+        public Task<T> GetPlain<T>(string aMethod, IList<IRequestParameter> aParams = null)
         {
-            return iExecutor.ExecuteGet(aMethod, RequestParameter.EmptyList)
+            if (aParams == null) aParams = RequestParameter.EmptyList;
+            return iExecutor.ExecuteGet(aMethod, aParams)
+
                             .ContinueWith((aTask) =>
                             {
                                 CheckHttpResponse(aTask);
+                                var msg = aTask.Result.Content.ReadAsStringAsync().Result;
```
Comment (projectgoav) [inline E.Deezer/DeezerClient.cs:141]: Likewise, I think `msg` here could go :)

File: E.Deezer/E.Deezer.csproj [hunk 4]
```diff
@@ -85,11 +85,9 @@
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="RequestParameter.cs" />
   </ItemGroup>
-  <ItemGroup />
   <ItemGroup>
-    <Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.10.0.3\lib\portable-net45+win8+wp8+wpa81\Newtonsoft.Json.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```
Comment (projectgoav) [inline E.Deezer/E.Deezer.csproj:89]: I'm not sure about this. I'd rather keep Newtonsoft at the 10.0.0 range. Iirc Newtonsoft stopped supporting PCL and moved .NET Standard in 11.0.0+. .NET Standard support is on my list of things to do if I ever get around to doing it. Thoughts?

File: E.Deezer/DeezerClient.cs [hunk 5]
```diff
@@ -244,6 +243,7 @@ internal Task Login()
...
```
Comment (projectgoav) [inline E.Deezer/DeezerClient.cs:246]: `msg` not required here.

**Review by projectgoav — CHANGES_REQUESTED** [review]
Code looks good :) A few things I'd like before I'd merge but I think they are pretty minor. More detail in the review comments.

- Captials for property names
- Remove the unused `msg` code
- Backout the Newtonsoft + System.Net.Http package upgrades

File: E.Deezer/DeezerClient.cs (hunk 2 above)
Comment (tolbxela) [reply E.Deezer/DeezerClient.cs:127]: sure, ist was for my debug. :-)

File: E.Deezer/E.Deezer.csproj (hunk 4 above)
Comment (tolbxela) [reply E.Deezer/E.Deezer.csproj:89]: I use the lib for my ASP.NET Core project, that's why I have updated the Newtonsoft.Json.
I know really nothing about PCL ... :-)

**Pull request closed without merging**